artman/cli/converter.py

```diff
@@ -0,0 +1,297 @@
+    result.common.api_name = legacy_common.get('api_name', 'unspecified')
+    result.common.api_version = legacy_common.get('api_version', 'unspecified')
+    result.common.organization_name = legacy_common.get(
+        'organization_name', 'unspecified')
+
+    # Python dict#get(key, default=value) doesn't well support list-type field.
+    # Instead of using defaultdict, simply do the if checking here.
+    if legacy_common.get('gapic_api_yaml'):
+        result.common.gapic_yaml = _sanitize_repl_var(
+            legacy_common.get('gapic_api_yaml')[0])
+    if legacy_common.get('service_yaml'):
+        result.common.service_yaml = _sanitize_repl_var(
+            legacy_common.get('service_yaml')[0])
+    if legacy_common.get('proto_deps'):
+        result.common.proto_deps.extend(
+            _compute_deps(legacy_common.get('proto_deps')))
+    if legacy_common.get('proto_test_deps'):
+        result.common.test_proto_deps.extend(
+            _compute_deps(legacy_common.get('proto_test_deps')))
+    if legacy_common.get('src_proto_path'):
+        result.common.src_proto_paths.extend(
+            _compute_src_proto_paths(legacy_common.get('src_proto_path')))
+
+    result.artifacts.extend(_compute_artifacts(legacy_config, legacy_common))
+    return result
+
+
+def _compute_artifacts(legacy_config, legacy_common):
+    result = []
+    LANGS = ['java', 'python', 'php', 'ruby', 'go', 'csharp', 'nodejs']
+
+    # Compute artifacts section
+    if legacy_common.get('git_repos'):
+        common_git_repos = legacy_common.get('git_repos')
+    else:
+        common_git_repos = []
+
+    for lang in LANGS:
+        if lang not in legacy_config:
+            continue
+        legacy_artifact_config = legacy_config[lang]
+        if legacy_artifact_config:
+            artifact = Artifact()
+            # Name the artifact as `{lang}_gapic`.
+            artifact.name = '%s_gapic' % lang
+            artifact.language = Artifact.Language.Value(lang.upper())
+
+            if 'release_level' in legacy_artifact_config:
+                artifact.release_level = Artifact.ReleaseLevel.Value(
+                    legacy_artifact_config.get('release_level').upper())
+
+            artifact.type = _compute_artifact_type(legacy_common)
+
+            # Compute package version.
+            if 'generated_package_version' in legacy_artifact_config:
+                legacy_package_version = legacy_artifact_config.get(
+                    'generated_package_version')
+                if 'lower' in legacy_package_version:
+                    artifact.package_version.grpc_dep_lower_bound = (
+                        legacy_package_version.get('lower'))
+                if 'upper' in legacy_package_version:
+                    artifact.package_version.grpc_dep_upper_bound = (
+                        legacy_package_version.get('upper'))
+
+            # Compute publishing targets
+            if 'git_repos' in legacy_artifact_config:
```

Comment:
If a lang doesn't define `git_repos`, doesn't it still automatically inherit the common git repo? (Maybe I am mistaken about this bit of existing artman logic, but it seems like all langs should stage to `api-client-staging` by default.) This looks like the new artifact will have no publish target if the legacy lang config doesn't define a git repo.